diff -ru Mesa-9.0.1/src/gallium/state_trackers/egl/wayland/native_shm.c Mesa-9.0.1-debug/src/gallium/state_trackers/egl/wayland/native_shm.c
--- Mesa-9.0.1/src/gallium/state_trackers/egl/wayland/native_shm.c	2012-12-21 15:12:15.980830067 +0000
+++ Mesa-9.0.1-debug/src/gallium/state_trackers/egl/wayland/native_shm.c	2012-12-21 20:52:03.569106298 +0000
@@ -164,8 +165,9 @@
       return FALSE;
 
    if (shmdpy->base.formats == 0)
-      wl_display_roundtrip(shmdpy->base.dpy);
-   if (shmdpy->base.formats == 0)
+      wayland_roundtrip(&shmdpy->base);
+
+  if (shmdpy->base.formats == 0)
       return FALSE;
 
    winsys = wayland_create_sw_winsys(shmdpy->base.dpy);

