diff --git a/src/plugins/platforms/wayland/wayland.pro b/src/plugins/platforms/wayland/wayland.pro
index 6639feb..2a74522 100644
--- a/src/plugins/platforms/wayland/wayland.pro
+++ b/src/plugins/platforms/wayland/wayland.pro
@@ -1,10 +1,10 @@
 TARGET = qwayland
-
-PLUGIN_TYPE = platforms
 load(qt_plugin)
 
 CONFIG += link_pkgconfig qpa/genericunixfontdatabase
 
+DESTDIR = $$QT.compositor.plugins/platforms
+
 QT += core-private gui-private platformsupport-private
 
 SOURCES =   main.cpp \
@@ -86,6 +86,9 @@ INCLUDEPATH += $$PWD/../../../shared
     LIBS += -lwayland-client
 }
 
+target.path += $$[QT_INSTALL_PLUGINS]/platforms
+INSTALLS += target
+
 include ($$PWD/gl_integration/gl_integration.pri)
 include ($$PWD/windowmanager_integration/windowmanager_integration.pri)
 
