diff --git a/src/multimedia/gsttools_headers/qgstxvimagebuffer_p.h b/src/multimedia/gsttools_headers/qgstxvimagebuffer_p.h
index 30cc4c9..559d867 100644
--- a/src/multimedia/gsttools_headers/qgstxvimagebuffer_p.h
+++ b/src/multimedia/gsttools_headers/qgstxvimagebuffer_p.h
@@ -59,6 +59,9 @@
 #include <QtCore/qwaitcondition.h>
 #include <QtCore/qqueue.h>
 
+#include <gst/gst.h>
+#include "qgstbufferpoolinterface_p.h"
+
 #include <X11/Xlib.h>
 #include <sys/ipc.h>
 #include <sys/shm.h>
@@ -67,9 +70,6 @@
 #include <X11/extensions/Xv.h>
 #include <X11/extensions/Xvlib.h>
 
-#include <gst/gst.h>
-#include "qgstbufferpoolinterface_p.h"
-
 QT_BEGIN_NAMESPACE
 
 class QGstXvImageBufferPool;
